models/protocol_tests/generate.go

```diff
@@ -356,15 +417,25 @@ func findMember(shape *api.Shape, key string) string {
 //
 // The shape's recursive values also will have assertions generated for them.
 func GenerateAssertions(out interface{}, shape *api.Shape, prefix string) string {
+	if shape == nil {
+		return ""
+	}
 	switch t := out.(type) {
 	case map[string]interface{}:
 		keys := util.SortedKeys(t)
 
 		code := ""
-		if shape.Type == "map" {
+		// If the shape is 'jsonvalue' we will begin iteration at the parent
+		// map[string]interface{}.
+		if shape.Type == "map" || shape.Type == "jsonvalue" {
 			for _, k := range keys {
 				v := t[k]
 				s := shape.ValueRef.Shape
+				/*if s == nil && shape.Type == "jsonvalue" {
```

Comment:
Still needed? Is there still a need to check for `jsonvalue` in `if shape.Type == "map" || shape.Type == "jsonvalue" {` now?